datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategyFactory.java

```diff
@@ -47,7 +52,15 @@ public AssignmentStrategy createStrategy(Properties assignmentStrategyProperties
         Optional.empty();
     Optional<Integer> partitionFullnessThresholdPct = cfgPartitionFullnessThresholdPct > 0 ?
         Optional.of(cfgPartitionFullnessThresholdPct) : Optional.empty();
+    String cluster = props.getString(CFG_CLUSTER_NAME);
+
+    // Create the ZooKeeper Client
+    String zkAddress = props.getString(CFG_ZK_ADDRESS);
+    int zkSessionTimeout = props.getInt(CFG_ZK_SESSION_TIMEOUT, ZkClient.DEFAULT_SESSION_TIMEOUT);
+    int zkConnectionTimeout = props.getInt(CFG_ZK_CONNECTION_TIMEOUT, ZkClient.DEFAULT_CONNECTION_TIMEOUT);
+    ZkClient zkClient = new ZkClient(zkAddress, zkSessionTimeout, zkConnectionTimeout);
```

Comment:
Also Creation of zkClient object will not go through if CFG_ZK_ADDRESS itself is not set. If I don't want to enable elastic task, I should not be forced to set CFG_ZK_ADDRESS.

Reply:
Only creating this object if `enableElasticTaskAssignment` is enabled.